wlauto/workloads/uxperf_applaunch/__init__.py

```diff
@@ -0,0 +1,227 @@
+        self.workload.uiauto_params[
+            'markers_enabled'] = self.workload.markers_enabled
+        self.workload.uiauto_params['applaunch_type'] = self.applaunch_type
+
+    def init_resources(self, context):
+        self.workload.init_resources(context)
+
+    def setup(self, context):
+        AndroidBenchmark.setup(self.workload, context)
+        if self.workload.launch_main is False:  # Required for launching skype
+            self.workload.launch_app()
+
+    def stop_application(self):
+        """kills the application process
+        """
+        self.device.execute('am force-stop {}'.format(self.workload.package))
+
+    def drop_pagecache(self):
+        """clears page cache
+        """
+        self.device.execute('sync')
+        self.device.execute('echo 1 > /proc/sys/vm/drop_caches', as_root=True)
+
+    def drop_inodecache(self):
+        """clears inode cache
+        """
+        self.device.execute('sync')
+        self.device.execute('echo 2 > /proc/sys/vm/drop_caches', as_root=True)
+
+    def drop_inodepagecache(self):
+        """clears inode and page caches
+        """
+        self.device.execute('sync')
+        self.device.execute('echo 3 > /proc/sys/vm/drop_caches', as_root=True)
+
+    def clean_application(self):
+        """Kills the application and clears caches if required
+        """
+        self.stop_application()
+        if(self.applaunch_type == 'launch_from_long-idle'):
+            self.drop_inodepagecache()
+
+    def update_iteration_result(self, context, i):
+        """Creates subfolders for every iterations and respective log files
+        """
+        iteration_folder = "{0}_{1}_{2}".format(
+            self.workload.name, i, self.applaunch_iterations)
+        iteration_path = os.path.join(
+            context.output_directory, iteration_folder)
+        if not os.path.exists(iteration_path):
+            os.makedirs(iteration_path)
+        logcat_log = os.path.join(iteration_path, i + '_logcat.log')
+        self.device.dump_logcat(logcat_log)
+
+    def update_result(self, context):
+        self.logcat_log = open(os.path.join(
+            context.output_directory, 'logcat.log'), 'w')
+        marker = 'UX_PERF'
+        for iter_dir in os.listdir(context.output_directory):
+            if self.workload.name in iter_dir:
+                dir_path = os.path.join(context.output_directory, iter_dir)
+                for log_file in os.listdir(dir_path):
+                    if log_file.endswith('.log'):
+                        log_number = "applaunch{}".format(
+                            log_file.split("_")[0])
+                        file_path = os.path.join(dir_path, log_file)
+                        with open(file_path, 'r') as log_data:
+                            for line in log_data.readlines():
+                                if marker in line and log_number in line:
+                                    self.logcat_log.write(line)
```

Comment:
Can part of this nested loop be split into a smaller method to make the code clearer? For example dealing with each log file.